src/Store/Bridge/MariaDB/Store.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Store\Bridge\MariaDB;
+
+use PhpLlm\LlmChain\Platform\Vector\Vector;
+use PhpLlm\LlmChain\Store\Document\Metadata;
+use PhpLlm\LlmChain\Store\Document\VectorDocument;
+use PhpLlm\LlmChain\Store\Exception\InvalidArgumentException;
+use PhpLlm\LlmChain\Store\InitializableStoreInterface;
+use PhpLlm\LlmChain\Store\VectorStoreInterface;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * Requires MariaDB >=11.7.
+ *
+ * @see https://mariadb.org/rag-with-mariadb-vector/
+ *
+ * @author Valtteri R <valtzu@gmail.com>
+ */
+final readonly class Store implements VectorStoreInterface, InitializableStoreInterface
+{
+    /**
+     * @param string $databaseName    The name of the database
+     * @param string $tableName       The name of the table
+     * @param string $indexName       The name of the vector search index
+     * @param string $vectorFieldName The name of the field int the index that contains the vector
+     */
+    public function __construct(
+        private \PDO $connection,
+        private string $databaseName,
```

Comment:
@chr-hertel what do you think about `$databaseName` – is it necessary? I think most of the time a connection has pre-selected database and passing this from Doctrine DSN etc could be difficult

I'm considering dropping this